tree/ntuple/v7/test/ntuple_zip.cxx

```diff
@@ -84,3 +84,134 @@ TEST(RNTupleZip, Large)
    decompressor.Unzip(zipBuffer.get(), szZip, N, unzipBuffer.get());
    EXPECT_EQ(data, std::string(unzipBuffer.get(), N));
 }
+
+TEST(RNTupleZip, CompressionOverride)
+{
+   RNTupleWriteOptions options;
+   EXPECT_EQ(404, options.GetCompression());
+   EXPECT_FALSE(options.IsCompressionOverride());
+   options.SetCompression(404);
+   EXPECT_TRUE(options.IsCompressionOverride());
+}
+
+TEST(RNTupleZip, TFilePtrCompressionSettings)
+{
+   // RNTuple added to a newly created TFile using the std::unique_ptr<TFile>&
+   // method will pick up changes to the TFile's compression settings
+   FileRaii fileGuard("test_ntuple_zip_tfileptr_comp.root");
+   {
+      std::unique_ptr<TFile> file = nullptr;
+      auto model = RNTupleModel::Create();
+      auto field = model->MakeField<float>("field");
+      auto klassVec = model->MakeField<std::vector<CustomStruct>>("klassVec");
+      RNTupleWriteOptions options;
+      options.SetCompression(404);
```

Comment:
Let's pick a more unusual compression setting that is more unlikely to be a default, e.g. 407

Reply:
Thank you, updated in 26d8cef